Lib/calendar.py

```diff
@@ -555,8 +555,6 @@ def __enter__(self):
         _locale.setlocale(_locale.LC_TIME, self.locale)
 
     def __exit__(self, *args):
-        if self.oldlocale is None:
-            return
```

Comment:
Can you explain this change?

A

Reply:
I explained it in the commit message. See a96786c.

Reply:
This was introduced in @vstinner's #31214 -- I agree with you though, `locale.setlocale` calls through to `_locale.setlocale` which appears to only return strings.

A

Reply:
I looked that PR up and believe these lines were added by mistake, following the common pattern of using the context manager. However, @vstinner mentions "corner cases" in https://github.com/python/cpython/pull/31214#issuecomment-1049863219, so maybe there is more to it. @vstinner can you comment?

In any case, if you think a96786c should be reverted, I'm fine with it. This is just a slight increase in the test coverage. BTW. if there is any corner case known, maybe we could cover it as well?

Reply:
I wrote the test to handle the case when ``self.oldlocale = _locale.setlocale(_locale.LC_TIME, None)`` fails in ``__enter__()``. I tested. It seems like ``__exit__()`` is not called if ``__enter__()`` raises an exception. So removing "oldlocale is None" test is correct.

Reply:
Thanks for the clarification! I'm marking this as resolved.

> It seems like __exit__() is not called if __enter__() raises an exception.

I believe this is part of the context manager protocol (`__exit__()` is only called if `__enter__()` has successfully returned the resource), however, I cannot find it in PEP-343 right now. But yeah, I'm pretty sure this is how it works since the `with` statement introduction in Python 2.5.